index.html

```diff
@@ -1,390 +1,1471 @@
+            this.container += chunk;
+            const lines = this.container.split('\r\n');
+            this.container = lines.pop();
+            lines.forEach(line => controller.enqueue(line));
+          }
+
+          flush(controller) {
+            controller.enqueue(this.container);
+          }
+        }
+
+        const lineReader = port.readable
+            .pipeThrough(new TextDecoderStream())
+            .pipeThrough(new TransformStream(new LineBreakTransformer()))
+            .getReader();
+      </pre>
+
+      Some other ways of encoding message boundaries are to prefix each message
+      with its length or to wait a defined length of time before transmitting
+      the next message. Implementing a {{TransformStream}} for these types of
+      message boundaries is left as an exercise for the reader.
+    </aside>
+
+    The {{SerialPort/readable}} getter steps are:
+
+    <ol>
+      <li>
+        If [=this=].{{[[readable]]}} is not `null`, return
+        [=this=].{{[[readable]]}}.
+      <li>If [=this=].{{[[state]]}} is not `"opened"`, return `null`.
+      <li>If [=this=].{{[[readFatal]]}} is `true`, return `null`.
+      <li>Let |stream| be a [=new=] {{ReadableStream}}.
+      <li>
+        Let |pullAlgorithm| be the following steps:
+        <ol>
+          <li>
+            Let |desiredSize| be the
+            <a href="https://streams.spec.whatwg.org/#desired-size-to-fill-a-streams-internal-queue">desired size</a>
+            of [=this=].{{[[readable]]}}'s
+            <a href="https://streams.spec.whatwg.org/#internal-queues">internal queue</a>.
+          <li>
+            Run the following steps [=in parallel=]:
+            <ol>
+              <li>
+                Invoke the operating system to read up to |desiredSize| bytes
+                from the port, putting the result in the [=byte sequence=]
+                |bytes|.
+              <li>
+                [=Queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=serial port task source=] to run the
+                following steps:
+                <ol>
+                  <li>
+                    If no errors were encountered run the following steps:
+                    <ol>
+                      <li>
+                        Let |buffer| be a [=new=] {{ArrayBuffer}} created from |bytes|.
+                      <li>
+                        Let |chunk| be a [=new=] {{Uint8Array}} view over |buffer|,
+                        who's length is the length of |bytes|.
+                      <li>
+                        Invoke [=ReadableStream/enqueue=] on
+                        [=this=].{{[[readable]]}} with |chunk|.
+                    </ol>
+                  <li>
+                    If a buffer overrun condition was encountered, invoke
+                    [=ReadableStream/error=] on [=this=].{{[[readable]]}} with a
+                    <dfn>`BufferOverrunError`</dfn> and invoke the steps to [=handle
```

Comment:
It should be clearer these are DOMExceptions. I'm not sure exactly how to mark that up given that there's no entry in the Web IDL table to link to yet, but the end result should look something like `"BufferOverrunError" DOMException` (as you've done for other DOMException instances).